src/evaluate.py

```diff
@@ -0,0 +1,72 @@
+import pandas as pd
+import numpy as np
+import re
+from src.model import yoModel
+from tqdm import tqdm
+from typing import List, Tuple
+from src.utils import WORDS_REGEX
+from dataclasses import dataclass
+from sklearn.metrics import accuracy_score, precision_score, recall_score, \
+    f1_score, roc_auc_score
+
+
+@dataclass
+class EvaluateResult:
+    """Model evaluation result."""
+    accuracy: float
+    precision: float
+    recall: float
+    f1_score: float
+    auroc: float
+
+    def __str__(self):
+        return "\n".join(["{}: {}".format(m, v) for m, v in self.__dict__.items()])
+
+
+def get_substrings(text: str) -> List[List[int]]:
+    """Function for get ye(yo) words indexes"""
+    res = []
+    for re_match in re.finditer(WORDS_REGEX, text):
+        if re.search(r'[ЕЁеё]', re_match.group()) is not None:
+            res.append(re_match.span())
+    return res
+
+
+def substrings2onehot(text:str, positions: List[Tuple[int, int]]) -> np.ndarray[bool]:
+    """Transform substring positions to one hot vector"""
+    words_pos = get_substrings(text)
+    one_hot_index = np.zeros(len(words_pos), dtype=bool)
+    for i, pos in enumerate(positions):
+        if pos in words_pos:
+            one_hot_index.put(i, True)
+
+    return one_hot_index
+
+
+def evaluate_model(model: yoModel, data: pd.DataFrame, verbose: bool=False) -> EvaluateResult:
```

Comment:
Выставляем отступы согласно [PEP8](https://peps.python.org/pep-0008/):
```suggestion
def evaluate_model(model: yoModel, data: pd.DataFrame, verbose: bool = False) -> EvaluateResult:
```